client/src/pages/Home.js

```diff
@@ -72,12 +72,41 @@ const Home = () => {
 	const handleChange = (setter) => (event) => {
 		setter(event.target.value);
 	};
+
 	const onOutputValueChange = handleChange(setOutputValue);
 	const onAlternativeValueChange = handleChange(setAlternativeValue);
 
-	const handleSubmit = (data) => {
-		setOutputValue(data);
-		setShowOutput(true);
+	const handleSubmit = () => {
+		if (inputValue.trim() !== "") {
+			setIsLoading(true); // Set loading state to true
+			fetchData();
+		}
+	};
+
+	const fetchData = async () => {
+		try {
+			const response = await fetch("/api/correction", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify({ input: inputValue }),
+			});
+
+			if (response.ok) {
+				const data = await response.json();
+				const correctedSentence = data;
+				setOutputValue(correctedSentence);
+				setShowOutput(true);
+			} else {
+				console.log("Error: " + response.status);
+			}
+		} catch (error) {
+			console.log(error);
+		} finally {
+			setIsLoading(false);
+			setInputValue("");
+		}
```

Comment:
The fetch is already done in `Input.js` component, it is not necessary to fetch again in here @hadipournigjeh

Reply:
Hi Lexi , unfortunately I made it this one and already it works , I don't have new ideas to change again :-(